### ISSUE TITLE
[Device Support Request] lumi.plug.maus01 (ZNCZ12LM)

### ISSUE THREAD

**Issue by ghost** [zigpy/zha-device-handlers#1654]
I have seen similar devices listed as issues here but not quite this exact model number ZNCZ12LM.
It is an [Aqara Smart Plug](https://www.amazon.com/gp/aw/d/B07CJ2MM6Z?psc=1&ref=ppx_pop_mob_b_asin_title). It will join to the network and can be controlled but much of the energy reporting metrics are showing as unavailable

<details>

<summary>Device signature</summary>

```yaml
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.Router: 1>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142>, manufacturer_code=4447, maximum_buffer_size=127, maximum_incoming_transfer_size=100, server_mask=0, maximum_outgoing_transfer_size=100, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=False, *is_full_function_device=True, *is_mains_powered=True, *is_receiver_on_when_idle=True, *is_router=True, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": 260,
      "device_type": "0x0051",
      "in_clusters": [
        "0x0000",
        "0x0001",
        "0x0002",
        "0x0003",
        "0x0004",
        "0x0005",
        "0x0006",
        "0x000a",
        "0x0010",
        "0x0b04"
      ],
      "out_clusters": [
        "0x000a",
        "0x0019"
      ]
    },
    "2": {
      "profile_id": 260,
      "device_type": "0x0009",
      "in_clusters": [
        "0x000c"
      ],
      "out_clusters": [
        "0x0004",
        "0x000c"
      ]
    },
    "3": {
      "profile_id": 260,
      "device_type": "0x0053",
      "in_clusters": [
        "0x000c"
      ],
      "out_clusters": [
        "0x000c"
      ]
    },
    "100": {
      "profile_id": 260,
      "device_type": "0x0107",
      "in_clusters": [
        "0x000f"
      ],
      "out_clusters": [
        "0x0004",
        "0x000f"
      ]
    }
  },
  "manufacturer": "LUMI",
  "model": "lumi.plug.maus01",
  "class": "zhaquirks.xiaomi.aqara.plug_maus01.Plug"
}
```

</details>

<details>

<summary>Diagnostic information</summary>

```yaml
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2022.7.4",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "aarch64",
    "timezone": "America/New_York",
    "os_name": "Linux",
    "os_version": "5.15.32-v8",
    "supervisor": "2022.07.0",
    "host_os": "Home Assistant OS 8.2",
    "docker_version": "20.10.14",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "alarmo": {
      "version": "v1.9.4",
      "requirements": []
    },
    "hacs": {
      "version": "1.26.2",
      "requirements": [
        "aiogithubapi\u003e\u003d22.2.4"
      ]
    },
    "noonlight": {
      "version": "1.0.2",
      "requirements": [
        "noonlight\u003d\u003d0.1.1"
      ]
    },
    "keymaster": {
      "version": "v0.0.82",
      "requirements": []
    },
    "scheduler": {
      "version": "v0.0.0",
      "requirements": []
    },
    "powercalc": {
      "version": "v0.22.1",
      "requirements": [
        "numpy\u003e\u003d1.21.1"
      ]
    }
  },
  "integration_manifest": {
    "domain": "zha",
    "name": "Zigbee Home Automation",
    "config_flow": true,
    "documentation": "https://www.home-assistant.io/integrations/zha",
    "requirements": [
      "bellows\u003d\u003d0.31.1",
      "pyserial\u003d\u003d3.5",
      "pyserial-asyncio\u003d\u003d0.6",
      "zha-quirks\u003d\u003d0.0.77",
      "zigpy-deconz\u003d\u003d0.18.0",
      "zigpy\u003d\u003d0.47.2",
      "zigpy-xbee\u003d\u003d0.15.0",
      "zigpy-zigate\u003d\u003d0.9.0",
      "zigpy-znp\u003d\u003d0.8.1"
    ],
    "usb": [
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*2652*",
        "known_devices": [
          "slae.sh cc2652rb stick"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*sonoff*plus*",
        "known_devices": [
          "sonoff zigbee dongle plus"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*tubeszb*",
        "known_devices": [
          "TubesZB Coordinator"
        ]
      },
      {
        "vid": "1A86",
        "pid": "7523",
        "description": "*zigstar*",
        "known_devices": [
          "ZigStar Coordinators"
        ]
      },
      {
        "vid": "1CF1",
        "pid": "0030",
        "description": "*conbee*",
        "known_devices": [
          "Conbee II"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8A2A",
        "description": "*zigbee*",
        "known_devices": [
          "Nortek HUSBZB-1"
        ]
      },
      {
        "vid": "0403",
        "pid": "6015",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate+"
        ]
      },
      {
        "vid": "10C4",
        "pid": "EA60",
        "description": "*zigate*",
        "known_devices": [
          "ZiGate"
        ]
      },
      {
        "vid": "10C4",
        "pid": "8B34",
        "description": "*bv 2010/10*",
        "known_devices": [
          "Bitron Video AV2010/10"
        ]
      }
    ],
    "codeowners": [
      "@dmulcahey",
      "@adminiuga"
    ],
    "zeroconf": [
      {
        "type": "_esphomelib._tcp.local.",
        "name": "tube*"
      },
      {
        "type": "_zigate-zigbee-gateway._tcp.local.",
        "name": "*zigate*"
      }
    ],
    "after_dependencies": [
      "onboarding",
      "usb",
      "zeroconf"
    ],
    "iot_class": "local_polling",
    "loggers": [
      "aiosqlite",
      "bellows",
      "crccheck",
      "pure_pcapy3",
      "zhaquirks",
      "zigpy",
      "zigpy_deconz",
      "zigpy_xbee",
      "zigpy_zigate",
      "zigpy_znp"
    ],
    "is_built_in": true
  },
  "data": {
    "ieee": "**REDACTED**",
    "nwk": 33253,
    "manufacturer": "LUMI",
    "model": "lumi.plug.maus01",
    "name": "LUMI lumi.plug.maus01",
    "quirk_applied": true,
    "quirk_class": "zhaquirks.xiaomi.aqara.plug_maus01.Plug",
    "manufacturer_code": 4447,
    "power_source": "Mains",
    "lqi": 255,
    "rssi": -60,
    "last_seen": "2022-07-15T00:04:34",
    "available": true,
    "device_type": "Router",
    "signature": {
      "node_descriptor": "NodeDescriptor(logical_type\u003d\u003cLogicalType.Router: 1\u003e, complex_descriptor_available\u003d0, user_descriptor_available\u003d0, reserved\u003d0, aps_flags\u003d0, frequency_band\u003d\u003cFrequencyBand.Freq2400MHz: 8\u003e, mac_capability_flags\u003d\u003cMACCapabilityFlags.AllocateAddress|RxOnWhenIdle|MainsPowered|FullFunctionDevice: 142\u003e, manufacturer_code\u003d4447, maximum_buffer_size\u003d127, maximum_incoming_transfer_size\u003d100, server_mask\u003d0, maximum_outgoing_transfer_size\u003d100, descriptor_capability_field\u003d\u003cDescriptorCapability.NONE: 0\u003e, *allocate_address\u003dTrue, *is_alternate_pan_coordinator\u003dFalse, *is_coordinator\u003dFalse, *is_end_device\u003dFalse, *is_full_function_device\u003dTrue, *is_mains_powered\u003dTrue, *is_receiver_on_when_idle\u003dTrue, *is_router\u003dTrue, *is_security_capable\u003dFalse)",
      "endpoints": {
        "1": {
          "profile_id": 260,
          "device_type": "0x0051",
          "in_clusters": [
            "0x0000",
            "0x0001",
            "0x0002",
            "0x0003",
            "0x0004",
            "0x0005",
            "0x0006",
            "0x000a",
            "0x0010",
            "0x0b04"
          ],
          "out_clusters": [
            "0x000a",
            "0x0019"
          ]
        },
        "2": {
          "profile_id": 260,
          "device_type": "0x0009",
          "in_clusters": [
            "0x000c"
          ],
          "out_clusters": [
            "0x0004",
            "0x000c"
          ]
        },
        "3": {
          "profile_id": 260,
          "device_type": "0x0053",
          "in_clusters": [
            "0x000c"
          ],
          "out_clusters": [
            "0x000c"
          ]
        },
        "100": {
          "profile_id": 260,
          "device_type": "0x0107",
          "in_clusters": [
            "0x000f"
          ],
          "out_clusters": [
            "0x0004",
            "0x000f"
          ]
        }
      }
    },
    "entities": [
      {
        "entity_id": "binary_sensor.lumi_lumi_plug_maus01_binary_input",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "button.lumi_lumi_plug_maus01_identify",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "sensor.lumi_lumi_plug_maus01_electrical_measurement_power_factor",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "sensor.lumi_lumi_plug_maus01_device_temperature",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "sensor.lumi_lumi_plug_maus01_analog_input",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "sensor.lumi_lumi_plug_maus01_analog_input_2",
        "name": "LUMI lumi.plug.maus01"
      },
      {
        "entity_id": "switch.lumi_lumi_plug_maus01_on_off",
        "name": "LUMI lumi.plug.maus01"
      }
    ],
    "neighbors": [],
    "endpoint_names": [
      {
        "name": "SMART_PLUG"
      },
      {
        "name": "MAIN_POWER_OUTLET"
      },
      {
        "name": "METER_INTERFACE"
      },
      {
        "name": "OCCUPANCY_SENSOR"
      }
    ],
    "user_given_name": "Living Room Wallflower",
    "device_reg_id": "3a508014a579c95a88a0e5fc1478d5a1",
    "area_id": "living_room",
    "cluster_details": {
      "1": {
        "device_type": {
          "name": "SMART_PLUG",
          "id": 81
        },
        "profile_id": 260,
        "in_clusters": {
          "0x0000": {
            "endpoint_attribute": "basic",
            "attributes": {
              "0x0004": {
                "attribute_name": "manufacturer",
                "value": "LUMI"
              },
              "0x0005": {
                "attribute_name": "model",
                "value": "lumi.plug.maus01"
              }
            },
            "unsupported_attributes": {}
          },
          "0x0001": {
            "endpoint_attribute": "power",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0002": {
            "endpoint_attribute": "device_temperature",
            "attributes": {
              "0x0000": {
                "attribute_name": "current_temperature",
                "value": 3800
              }
            },
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0003": {
            "endpoint_attribute": "identify",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0006": {
            "endpoint_attribute": "on_off",
            "attributes": {
              "0x0000": {
                "attribute_name": "on_off",
                "value": 1
              }
            },
            "unsupported_attributes": {}
          },
          "0x0005": {
            "endpoint_attribute": "scenes",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0010": {
            "endpoint_attribute": "binary_output",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x000a": {
            "endpoint_attribute": "time",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0b04": {
            "endpoint_attribute": "electrical_measurement",
            "attributes": {
              "0x0304": {
                "attribute_name": "total_active_power",
                "value": 0.009568437933921814
              },
              "0x050b": {
                "attribute_name": "active_power",
                "value": 1.9700000286102295
              }
            },
            "unsupported_attributes": {
              "0x0300": {
                "attribute_name": "ac_frequency"
              },
              "0x0302": {
                "attribute_name": "ac_frequency_max"
              },
              "0x0505": {
                "attribute_name": "rms_voltage"
              },
              "0x0508": {
                "attribute_name": "rms_current"
              },
              "0x050a": {
                "attribute_name": "rms_current_max"
              },
              "0x050b": {
                "attribute_name": "active_power"
              },
              "0x050d": {
                "attribute_name": "active_power_max"
              },
              "0x0507": {
                "attribute_name": "rms_voltage_max"
              },
              "0x050f": {
                "attribute_name": "apparent_power"
              }
            }
          }
        },
        "out_clusters": {
          "0x0019": {
            "endpoint_attribute": "ota",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x000a": {
            "endpoint_attribute": "time",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      },
      "2": {
        "device_type": {
          "name": "MAIN_POWER_OUTLET",
          "id": 9
        },
        "profile_id": 260,
        "in_clusters": {
          "0x000c": {
            "endpoint_attribute": "analog_input",
            "attributes": {
              "0x0055": {
                "attribute_name": "present_value",
                "value": 1.9199999570846558
              }
            },
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x000c": {
            "endpoint_attribute": "analog_input",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      },
      "3": {
        "device_type": {
          "name": "METER_INTERFACE",
          "id": 83
        },
        "profile_id": 260,
        "in_clusters": {
          "0x000c": {
            "endpoint_attribute": "analog_input",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x000c": {
            "endpoint_attribute": "analog_input",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      },
      "100": {
        "device_type": {
          "name": "OCCUPANCY_SENSOR",
          "id": 263
        },
        "profile_id": 260,
        "in_clusters": {
          "0x000f": {
            "endpoint_attribute": "binary_input",
            "attributes": {},
            "unsupported_attributes": {}
          }
        },
        "out_clusters": {
          "0x000f": {
            "endpoint_attribute": "binary_input",
            "attributes": {},
            "unsupported_attributes": {}
          },
          "0x0004": {
            "endpoint_attribute": "groups",
            "attributes": {},
            "unsupported_attributes": {}
          }
        }
      }
    }
  }
}
```

</details>

<details>
<summary>Additional logs</summary>

```
Paste any additional debug logs here.
Don't remove the extra line breaks outside the ``` marks.
```

</details>

**Additional context**

**Comment by multicm** [comment]
I updated to 2022.7.5 (zigpy 0.47.3) removed the device and re-paired it and the issue persists

**Comment by TheJulianJES** [comment]
I must have opened my issue just after yours. I'm working on a fix but it won't be implemented officially for quite a while probably (as it would break other MAEU01 plugs).
It should be fairly easy to provide you with some custom quirk download links for now (they should work but could break with future HA updates).
See https://github.com/zigpy/zha-device-handlers/issues/1655 and https://github.com/zigpy/zha-device-handlers/pull/1656
I'll update you when the custom quirks are ready to be tested.

**Comment by TheJulianJES** [comment]
@multicm You can see the ``Testing`` section of this PR: https://github.com/zigpy/zha-device-handlers/pull/1656
Let me know if this also works for US plugs.

**Comment by multicm** [comment]
> I must have opened my issue just after yours. I'm working on a fix but it won't be implemented officially for quite a while probably (as it would break other MAEU01 plugs). It should be fairly easy to provide you with some custom quirk download links for now (they should work but could break with future HA updates). See #1655 and #1656 I'll update you when the custom quirks are ready to be tested.
> 
> Edit: Actually, your signature mentions a `lumi.plug.maus01` US plug and not an EU plug like mentioned in the title. So I'm not sure this will work for you
> 
> Edit 2: Maybe the quirk could fix stuff for you too

I apologize, yes I am looking for the US plug fix, that was a typo in the title. I just fixed it.

**Comment by TheJulianJES** [comment]
> I am going to test something on Monday. I think my issue is with the basic zigbee integration with home assistant only allowing 1 sensor per device (I read this somewhere...)

I think this should no longer be an issue. If I'm correct, the issue is that ZHA (the integration in HA) creates no sensor because the (locally/faked) ``active_power`` didn't have a default value without the changes in the custom quirk.

​

> so I am getting a conbee on Monday and will try it with zigbee2mqtt and see if that works.

You can try that. It's also possible that the plug behaves differently depending on coordinator firmware (seen that with the MAEU01). Also note that if you want to switch Z2M in the long-term, use a CC2652 based stick. The Conbee isn't really recommended to use with ZHA nor Z2M.

**Comment by multicm** [comment]
> @multicm You can see the `Testing` section of this PR: #1656 Let me know if this also works for US plugs.

I tested this but didn't see any change. I followed the included instructions.

There was one minor change in that volts appear now, but they stay 0

**Comment by TheJulianJES** [comment]
Don't think this would fix it either, but there's likely some new firmware available at the link [here](https://github.com/Koenkk/zigbee-OTA/blob/74d6c15b14a78319eb086f2c215f38c71d572571/index.json#L1827).

**Comment by dereitz** [comment]
Has anyone had success with MAUS1 plugs?  I'm particularly interested in the ability to set the power memory option.

@TheJulianJES if you're willing to work with me, I'd be happy to test.  Do you still recommend trying the update you provided back in Jul?

**Comment by TheJulianJES** [comment]
Due to some issues with newer Aqara firmware versions on the EU plugs with IKEA remotes, I don't recommend updating the firmware for now.
@dereitz Just to confirm, your model also reports as a ``lumi.plug.maus01`` (and not simply a ``lumi.plug``)?

If so, I've made some test changes which could make the power memory option work.
To test these changes:
- update to Home Assistant Core 2022.10.5 (or later)
- configure a ZHA custom quirk folder
- download this file: https://raw.githubusercontent.com/TheJulianJES/zha-device-handlers/tjj/xiaomi_us_plug_fixes/zhaquirks/xiaomi/aqara/plug_maus01.py
- put the downloaded ``plug_maus01.py`` file in the ``zha_custom_quirks`` folder (or in a sub folder)
- restart HA
- see on the device page if the custom quirk was applied (official quirk always starts with ``zhaquirks.xiaomi.aqara`` and the custom quirk should be much shorter depending on how the file/(sub-)folder is named)
- test what sensors work
- (possibly remove and) re-pair the plug and see if further sensors work now
- test the ``power_outage_memory`` attribute:

Open the ZHA device page, click on the three dots, click "Manage Zigbee Device" and select the ``OppleCluster`` and ``power_outage_memory``:
![image](https://user-images.githubusercontent.com/6409465/197087101-fd2adac8-603d-4176-b20e-2253e7d6faa8.png)
Then, click "Read Attribute" and see if it returns ``Bool.false``. If it does, remove that from the field and put a ``1`` in there and click ``Write Attribute``.
Read the attribute again and see if it now returns ``Bool.true``. (To reset the changes: use a ``0`` for writing to disable the function again).

(Always leave the second "Manufacturer Code Override" field completely empty (like in my screenshot))

**Comment by dereitz** [comment]
Wow, thank you so much for the quick, detailed response!!!

> Due to some issues with newer Aqara firmware versions on the EU plugs with IKEA remotes, I don't recommend updating the firmware for now. @dereitz Just to confirm, your model also reports as a `lumi.plug.maus01` (and not simply a `lumi.plug`)?

Correct, mine shows as `lumi.plug.maus01`

]> If so, I've made some test changes which could make the power memory option work. To test these changes:
> 
>     * update to Home Assistant Core 2022.10.5 (or later) 
>     * configure a ZHA custom quirk folder
>     * download this file: https://raw.githubusercontent.com/TheJulianJES/zha-device-handlers/tjj/xiaomi_us_plug_fixes/zhaquirks/xiaomi/aqara/plug_maus01.py
>     * put the downloaded `plug_maus01.py` file in the `zha_custom_quirks` folder (or in a sub folder)
>     * restart HA
>     * see on the device page if the custom quirk was applied (official quirk always starts with `zhaquirks.xiaomi.aqara` and the custom quirk should be much shorter depending on how the file/(sub-)folder is named)

All confirmed, I'm running 2022.10.5, I enabled the ZHA custom folder, placed the file in a subfolder, restarted, and I now see `Quirk: plug_maus01.Plug` once I expand "Zigbee info" under the Device Info section.

>     * test what sensors work

I get the following sensors:
![image](https://user-images.githubusercontent.com/1048973/197095081-0d743bf0-1d58-44d7-9453-96003b1a7dde.png)


>     * (possibly remove and) re-pair the plug and see if further sensors work now

After removing and re-pairing, I see the following:
![image](https://user-images.githubusercontent.com/1048973/197095638-1d2312db-4de8-4d47-9416-73151ab5a1f7.png)

>     * test the `power_outage_memory` attribute:
> 
> 
> Open the ZHA device page, click on the three dots, click "Manage Zigbee Device" and select the `OppleCluster` and `power_outage_memory`: ![image](https://user-images.githubusercontent.com/6409465/197087101-fd2adac8-603d-4176-b20e-2253e7d6faa8.png) Then, click "Read Attribute" and see if it returns `Bool.false`. If it does, remove that from the field and put a `1` in there and click `Write Attribute`. Read the attribute again and see if it now returns `Bool.true`. (To reset the changes: use a `0` for writing to disable the function again).
> 
> (Always leave the second "Manufacturer Code Override" field completely empty (like in my screenshot))

When I try to read, it simply says "None".  When I enter "1" and Write Attribute, it gives me a checkmark, but when I go to read the value again, it says "None".  I never get any Bool.true or Bool.false results at all.

Thanks again for the help!!!

**Comment by TheJulianJES** [comment]
Ah, that's a bummer. Thanks for all the screenshots though! So far, it seems like the firmware running on those plugs seems to differ quite a bit from the EU plug firmware. It's possible that a firmware update could help but I doubt it so far.
Zigbee2MQTT also doesn't seem to support "power outage memory" for the ``lumi.plug.maus01`` (but does support it for the ``lumi.plug``. It's a completely different attribute compared to the EU plug one there)

I don't think there's much I can do at the moment. It's possible that the total power consumption/smart metering could be fixed when the second AnalogInput cluster is properly replaced with another one that relays the info on the consumption bus (to a fake (Smart) MeteringCluster) but I don't think power outage memory can be added, as it doesn't seem to be supported in firmware at all.

I'm interested in how old the firmware is though:
Could you read ``app_version``, ``date_code`` (and ``sw_build_id``) from the ``Basic`` cluster and post the responses here?

**Comment by dereitz** [comment]
> I'm interested in how old the firmware is though: Could you read `app_version`, `date_code` (and `sw_build_id`) from the `Basic` cluster and post the responses here?

Absolutely!  Here you go:

app_version: 17
date_code: 12-24-2021
sw_build_id: None

**Comment by Rogue136198** [comment]
I also have a few of these devices. With your custom quirk I have them working for energy monitoring but I am trying to get the default power state set. What can I do to help get this working?

**Comment by dereitz** [comment]
I ended up returning my Aqara.  I found that the Sengled plugs work out of the box and are cheaper.  Good luck!

**Comment by Rogue136198** [comment]
> I ended up returning my Aqara. I found that the Sengled plugs work out of the box and are cheaper. Good luck!

Okay, Good to know. I'll give the Sengled ones a try.

**Comment by TheJulianJES** [comment]
This should partly be fixed in Home Assistant [2023.1.0b5](https://github.com/home-assistant/core/releases/tag/2023.1.0b5) and later by:
- https://github.com/zigpy/zha-device-handlers/pull/2057

You might need to fully remove the plug and re-pair it to get the functionality working.

Note that the issue also states: "There are several other functions supported by the device that I'll add in a subsequent PR." 

Also, this is likely a duplicate of https://github.com/zigpy/zha-device-handlers/issues/1636
